src/object-to-ast.ts

```diff
@@ -1,32 +1,44 @@
-module.exports = objectToAST
+import type babelCore from '@babel/core'
 
-function objectToAST(object, {babel, fileOptions}) {
+type ObjectToASTOptions = {
+  babel: typeof babelCore
+  fileOptions: babelCore.TransformOptions
+}
+
+export default function objectToAST(
+  object: unknown,
+  {babel, fileOptions}: ObjectToASTOptions,
+) {
   const stringified = stringify(object)
   const variableDeclarationNode = babel.template(`var x = ${stringified}`, {
     preserveComments: true,
     placeholderPattern: false,
     ...fileOptions.parserOpts,
     sourceType: 'module',
-  })()
+  })() as babelCore.types.VariableDeclaration
```

Comment:
I worked out where `declarations` is defined and it makes sens based on what is being passed the `babel.template()` (i.e. a declaration of `var x ...`).

I went for a straight up cast here given the template is essentially hard coded to be a `VariableDeclaration`, but I've got the voice in the back of my head telling me this is wrong and I should use a type guard.

This type guard works:

```ts
function isVariableDeclaration(node: babelCore.types.Statement | babelCore.types.Statement[]): node is babelCore.types.VariableDeclaration {
  return !Array.isArray(node) && node.type === 'VariableDeclaration'
}
```

However, I'm not sure what code to put on the failing side of it:

```ts
  if (isVariableDeclaration(variableDeclarationNode)) {
    return variableDeclarationNode.declarations[0].init // knows the type now without casting :)
  }

  // what now? Error? You can't get here...
```